projects/unlikelihood/agents.py

```diff
@@ -0,0 +1,866 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import math
+import torch
+import torch.nn.functional as F
+import json
+from collections import defaultdict, Counter
+from nltk import ngrams
+
+from parlai.utils.misc import round_sigfigs
+from parlai.agents.transformer.transformer import TransformerGeneratorAgent
+from parlai.core.torch_agent import TorchAgent, Batch, Output
+
+from parlai_internal.agents.parlall.agents import ParlallAgent
```

Comment:
@stephenroller  I don't think parlall is open sourced yet

Reply:
It is, it goes by the name image_seq2seq. We can strip it for now I think.